### PR TITLE
Gi 165 ergebnisse und details

### PR DESCRIPTION
**Pull request by DamarisLi**
## Description

This PR updates the results and details pages for study programmes. The changes replace mock data with real database queries and add more detailed information about a specific study programme in the details page.

## Jira Ticket

Link to the Jira ticket: [GI-165](https://techstudyfinder.atlassian.net/browse/GI-165)

## Author Checklist

- [ ] Code follows project coding standards and conventions
- [ ] Tests have been written or updated
- [ ] All tests pass locally
- [ ] No sensitive data or secrets are included
- [ ] Documentation (README, comments) has been updated if necessary

## Reviewer Notes

Please verify the following during review:

- The implementation matches the described requirements and Jira ticket
- Code changes are logically correct and maintainable
- No unnecessary dependencies or side effects introduced
- Tests cover critical logic and edge cases
- Naming, structure, and readability meet team standards

## Additional Context (optional)

Add any background information or references that help reviewers understand the changes.

---

_Please ensure all criteria are met before merging._

### CODE REVIEW

File: server/src/types/studyProgramme.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+export interface StudyProgramme {
```
Comment (psophis) [inline server/src/types/studyProgramme.ts]: Hier sind viele types nullable, im client aber nicht. Das könnte Probleme geben, wenn das Backend null-Werte akzeptiert, der Client aber nicht

File: server/src/repositories/quiz.repository.ts [hunk 2]
```diff
@@ -147,3 +148,18 @@ export async function getQuestionsLevel2(): Promise<any[]> {
...
+  const result = await pool.query(
+    "SELECT * FROM studiengang_full_view WHERE studiengang_id = $1",
+    [id],
+  );
+
+  if (result.rows.length === 0) {
+    return undefined;
+  }
+
+  return result.rows[0];
```
[the remark this answers is not in the document]
Comment (DamarisLi) [reply server/src/repositories/quiz.repository.ts]: Für welche Methode ist das? getStudyProgrammeById?

[the remark this answers is not in the document]
Comment (DamarisLi) [reply server/src/repositories/quiz.repository.ts]: Also der Methode wird ja die ID übergeben, deswegen verstehe ich nicht wieso sie die ID wieder returnen sollte :D

**Comment by johanna2402** [comment]
Bei sehr langen Schwerpunkten werden die nicht komplett angezeigt:
Bildschirm­foto 2026-01-07 um 14 27 59

Das müsste denke ich noch geändert werden
